### PR TITLE
ROX-24471: Set Matcher's memory req/lim aligned with consumption

### PR DESCRIPTION
**Pull request by jvdm**
### Description

Matcher's memory consumption and CPU consumption dropped significantly after #11168. This change updates the memory requirements to align with what is being observed in our production instances, example:

cpu
mem

And a wider look at the time when #11168 was deployed:

cpu
mem

### User-facing documentation

(*must be* 2 items and both *must be* checked)

- [x] CHANGELOG is updated
- [ ] CHANGELOG update is not needed
- [ ] Documentation PR is created and linked above
- [x] Documentation is not needed

### Testing

- [ ] inspected CI results

#### Automated testing

N/A

#### How I validated my change

1. Build helm templates.
2. Deploy Central cluster.
3. Verify resources:
   ```sh
   kubectl -n stackrox get deployment/scanner-v4-matcher -o yaml | yq '.spec.template.spec.containers[] | select(.name == "matcher") | .resources'
   limits:
     cpu: "2"
     memory: 2Gi
   requests:
     cpu: "1"
     memory: 500Mi
   ```

### CODE REVIEW

File: image/templates/helm/stackrox-central/internal/defaults.yaml.htpl [hunk 1]
```diff
@@ -259,10 +259,10 @@ defaults:
                 operator: DoesNotExist
       resources:
         requests:
-          memory: "4Gi"
+          memory: "500Mi"
```
Comment (daynewlee) [inline image/templates/helm/stackrox-central/internal/defaults.yaml.htpl:262]: LGTM. Do we need to set to 1G just in case?

**Review by daynewlee — APPROVED** [review]

File: image/templates/helm/stackrox-central/internal/defaults.yaml.htpl (hunk 1 above)
Comment (RTann) [reply image/templates/helm/stackrox-central/internal/defaults.yaml.htpl:262]: Seems like a single instance is only maxing at most 200Mi, so I think 500Mi is good

**Review by RTann — APPROVED** [review]

File: image/templates/helm/stackrox-central/internal/defaults.yaml.htpl (hunk 1 above)
Comment (RTann) [reply image/templates/helm/stackrox-central/internal/defaults.yaml.htpl:262]: Also note we set the Go runtime soft memory limit to 90% of the request, so it'll be 450Mi. Again it's a soft limit, but if set too low may cause GC thrashing. Again based on the graphs, seems like they max at 200Mi, so this should be sufficient

**Review by mclasmeier — APPROVED** [review]

File: image/templates/helm/stackrox-central/internal/defaults.yaml.htpl (hunk 1 above)
Comment (ludydoo) [reply image/templates/helm/stackrox-central/internal/defaults.yaml.htpl:262]: If the instance is maxing at 200Mi, shouldn't 200Mi be the `limit` not the `request`?

Comment (ludydoo) [reply image/templates/helm/stackrox-central/internal/defaults.yaml.htpl:262]: This looks about right. Hovering at 750Mi
Screenshot 2024-07-03 at 10 19 29 AM

Comment (jvdm) [reply image/templates/helm/stackrox-central/internal/defaults.yaml.htpl:262]: > If the instance is maxing at 200Mi, shouldn't 200Mi be the limit not the request?

We are being conservative; We don't know yet what can cause spikes in the Matcher's memory consumption at scale. And the improvements done in the vulnerability updater are localized to writing vulnerabilities. Keeping the limit high now so it won't unnecessarily OOM, while requiring what we believe is enough for a healthy Matcher.

**Review by dcaravel — APPROVED** [review]

**Pull request merged**